### PR TITLE
fix knative downstream tests

### PR DESCRIPTION
**Pull request by imjasonh**
Signed-off-by: Jason Hall <jason@chainguard.dev>

/assign vdemeester

cc @dprotaso @cardil 

# Changes

This moves the temp gopath to a location outside the repo, so that knative's `go-ls-tags` doesn't consider it when determining which tags to build for.

/kind cleanup

Might fix https://github.com/knative/actions/issues/86

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

PR friction log:
- unit test flake (timeout didn't happen): https://prow.tekton.dev/view/gs/tekton-prow/pr-logs/pull/tektoncd_pipeline/5763/pull-tekton-pipeline-unit-tests/1593602003307597824
- e2e test flake (ghcr.io failure trying to publish images): https://prow.tekton.dev/view/gs/tekton-prow/pr-logs/pull/tektoncd_pipeline/5763/pull-tekton-pipeline-alpha-integration-tests/1593602003462787072
- unit test flake (cloud events): https://prow.tekton.dev/view/gs/tekton-prow/pr-logs/pull/tektoncd_pipeline/5763/pull-tekton-pipeline-unit-tests/1593634438405689344

### CODE REVIEW

File: hack/setup-temporary-gopath.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ set -o nounset
 # the current repo directory is not within GOPATH.
 function shim_gopath() {
   local REPO_DIR=$(git rev-parse --show-toplevel)
-  local TEMP_GOPATH="${REPO_DIR}/.gopath"
+  local TEMP_GOPATH="/tmp/.gopath"
```
Comment (dprotaso) [inline hack/setup-temporary-gopath.sh:11]:
```suggestion
  local TEMP_GOPATH="$(mktemp -d)/.gopath"
```

Comment (dprotaso) [reply hack/setup-temporary-gopath.sh:11]: and in other places

Comment (vdemeester) [reply hack/setup-temporary-gopath.sh:11]: Same comment as @dprotaso, using `mktemp` is "better" 👼🏼

**Review by vdemeester — APPROVED** [review]

**Pull request merged**